_shared/project/tox.ini

```diff
@@ -64,27 +64,36 @@ deps =
         {{- include("tox/deps", indent=4) -}}
     {% endif %}
 depends =
-{% if cookiecutter._directory == 'pyramid-app' %}
+{% if cookiecutter._directory in ['pyapp', 'pyramid-app'] %}
     coverage: tests
 {% else %}
     coverage: tests,py{{ "{" }}{{ python_versions()|rest|pyformat(PyFormats.MAJOR_MINOR_FMT)|separator(",") }}{{ "}" }}-tests
 {% endif %}
+{% if cookiecutter.get("db") == "yes" or include_exists("tox/allowlist_externals") %}
+allowlist_externals =
+{% if cookiecutter.get("db") == "yes" %}
+    tests,functests: sh
+{% endif %}
```

Comment:
I could maybe simplify the template here by just allowing `sh` unconditionally even though it's only really needed when `db` is on.